### ISSUE TITLE
Finding installed ssh failed: spawn ENOTDIR

### ISSUE THREAD

**Issue by initial-wu** [microsoft/vscode-remote-release#4630]
- VS Code Version: 1.54.1
- OS Version: macOS Big Sur

Steps to Reproduce:

1. update VS Code to 1.54.1
2. connect to the ssh remote server

```
[12:19:55.078] Log Level: 2
[12:19:55.079] remote-ssh@0.65.0
[12:19:55.079] darwin x64
[12:19:55.080] SSH Resolver called for "ssh-remote+7b22686f73744e616d65223a22646576436c6f7564227d", attempt 1
[12:19:55.080] "remote.SSH.useLocalServer": true
[12:19:55.080] "remote.SSH.sshPath": undefined
[12:19:55.080] "remote.SSH.sshConfigurationFile": undefined
[12:19:55.080] "remote.SSH.useFlock": true
[12:19:55.080] "remote.SSH.lockfilesInTmp": false
[12:19:55.081] "remote.SSH.localServerDownload": auto
[12:19:55.081] "remote.SSH.remoteServerListenOnSocket": false
[12:19:55.081] "remote.SSH.showLoginTerminal": false
[12:19:55.081] "remote.SSH.defaultExtensions": []
[12:19:55.081] SSH Resolver called for host: devCloud
[12:19:55.081] Setting up SSH remote "devCloud"
[12:19:55.083] Acquiring local install lock: /var/folders/33/rdh80fdx5tx9dllx9pm_rgh80000gn/T/vscode-remote-ssh-devCloud-install.lock
[12:19:55.089] Looking for existing server data file at /Users/initialwu/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-devCloud-f30a9b73e8ffc278e71575118b6bf568f04587c8-0.65.0/data.json
[12:19:55.089] Using commit id "f30a9b73e8ffc278e71575118b6bf568f04587c8" and quality "stable" for server
[12:19:55.091] Install and start server if needed
[12:19:55.094] Checking ssh with "ssh -V" /
[12:19:55.096] Finding installed ssh failed: spawn ENOTDIR
[12:19:55.099] Resolver error: Error: spawn ENOTDIR
	at ChildProcess.spawn (internal/child_process.js:407:11)
	at Object.spawn (child_process.js:557:9)
	at /Users/initialwu/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.0/out/extension.js:1:141963
	at new Promise (<anonymous>)
	at k (/Users/initialwu/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.0/out/extension.js:1:141856)
	at I (/Users/initialwu/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.0/out/extension.js:1:141807)
	at /Users/initialwu/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.0/out/extension.js:1:145955
	at _ (/Users/initialwu/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.0/out/extension.js:1:146020)
	at C (/Users/initialwu/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.0/out/extension.js:1:141586)
	at Object.t.getSshCommandPath (/Users/initialwu/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.0/out/extension.js:1:144740)
	at Function.create (/Users/initialwu/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.0/out/extension.js:1:262422)
	at async Object.t.tryInstallWithLocalServer (/Users/initialwu/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.0/out/extension.js:1:387826)
	at async /Users/initialwu/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.0/out/extension.js:1:293745
	at async Object.t.withShowDetailsEvent (/Users/initialwu/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.0/out/extension.js:1:407000)
	at async /Users/initialwu/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.0/out/extension.js:1:387151
	at async S (/Users/initialwu/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.0/out/extension.js:1:383919)
	at async Object.t.resolveWithLocalServer (/Users/initialwu/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.0/out/extension.js:1:386778)
	at async Object.t.resolve (/Users/initialwu/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.0/out/extension.js:1:297757)
	at async /Users/initialwu/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.0/out/extension.js:127:110485
[12:19:55.101] ------
```

Does this issue occur when all extensions are disabled?: No

**Comment by roblourens** [comment]
Can anyone check whether setting `"remote.SSH.useLocalServer": false` makes any difference?

**Comment by elisakreiss** [comment]
> Can anyone check whether setting `"remote.SSH.useLocalServer": false` makes any difference?

This doesn't make a difference for me. Still the same error.

**Comment by roblourens** [comment]
And to clarify, this issue is only for getting this while trying to connect with Remote-SSH, and seeing the log in the original post. Getting this while opening a Jupyter notebook: make sure you've gotten the 1.54.1 update, and if so, open a new issue or  check https://github.com/microsoft/vscode/issues/118124.

**Comment by deepak1556** [comment]
@initial-wu or anyone on facing this issue, can you print your `PATH` value.

**Comment by elisakreiss** [comment]
> And to clarify, this issue is only for getting this while trying to connect with Remote-SSH, and seeing the log in the original post. Getting this while opening a Jupyter notebook: make sure you've gotten the 1.54.1 update, and if so, open a new issue or check [microsoft/vscode#118124](https://github.com/microsoft/vscode/issues/118124).
> 
> If you think you are seeing the same SSH issue, also please include your log from the `Remote-SSH` output channel
> 
> @initial-wu this line in your log: `[12:19:55.094] Checking ssh with "ssh -V" /` I am not sure how that slash would show up. Is it possible that was an accidental keypress from you, or does that actually appear in the log?

Is this helpful?
```[16:52:26.114] Log Level: 2
[16:52:26.116] remote-ssh@0.65.1
[16:52:26.116] darwin x64
[16:52:26.117] SSH Resolver called for "ssh-remote+7b22686f73744e616d65223a226e6f646531332d63636e636c75737465722e7374616e666f72642e656475222c2275736572223a22656b7265697373227d", attempt 1
[16:52:26.118] "remote.SSH.useLocalServer": false
[16:52:26.118] "remote.SSH.showLoginTerminal": false
[16:52:26.119] "remote.SSH.remotePlatform": {}
[16:52:26.119] "remote.SSH.sshPath": undefined
[16:52:26.119] "remote.SSH.sshConfigurationFile": undefined
[16:52:26.119] "remote.SSH.useFlock": true
[16:52:26.119] "remote.SSH.lockfilesInTmp": false
[16:52:26.119] "remote.SSH.localServerDownload": auto
[16:52:26.120] "remote.SSH.remoteServerListenOnSocket": false
[16:52:26.120] "remote.SSH.showLoginTerminal": false
[16:52:26.120] "remote.SSH.defaultExtensions": []
[16:52:26.120] SSH Resolver called for host: user@server.edu
[16:52:26.120] Setting up SSH remote "server.edu"
[16:52:26.195] Using commit id "f30a9b73e8ffc278e71575118b6bf568f04587c8" and quality "stable" for server
[16:52:26.216] Install and start server if needed
[16:52:28.807] Checking ssh with "ssh -V"
[16:52:28.810] Finding installed ssh failed: spawn ENOTDIR
[16:52:28.817] Resolver error: Error: spawn ENOTDIR
	at ChildProcess.spawn (internal/child_process.js:407:11)
	at Object.spawn (child_process.js:557:9)
	at /Users/user/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.1/out/extension.js:1:141985
	at new Promise (<anonymous>)
	at k (/Users/user/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.1/out/extension.js:1:141906)
	at I (/Users/user/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.1/out/extension.js:1:141857)
	at /Users/user/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.1/out/extension.js:1:145959
	at _ (/Users/user/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.1/out/extension.js:1:146024)
	at C (/Users/user/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.1/out/extension.js:1:141636)
	at R (/Users/user/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.1/out/extension.js:1:141545)
	at Object.t.generateMultiLineCommand (/Users/user/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.1/out/extension.js:1:143177)
	at z (/Users/user/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.1/out/extension.js:1:297373)
	at /Users/user/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.1/out/extension.js:1:294263
	at processTicksAndRejections (internal/process/task_queues.js:97:5)
	at async Object.t.withShowDetailsEvent (/Users/user/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.1/out/extension.js:1:407055)
	at async Object.t.resolve (/Users/user/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.1/out/extension.js:1:297912)
	at async /Users/user/.vscode/extensions/ms-vscode-remote.remote-ssh-0.65.1/out/extension.js:127:110485
[16:52:28.859] ------

**Comment by LittleMaer** [comment]
i meet the same issue

**Comment by roblourens** [comment]
We are able to reproduce it - if your PATH variable includes a path to a file, then this will be triggered.

Looking into a fix, but in the meantime, check that your PATH includes only folders, not files.

**Comment by omeraloni** [comment]
I don't have any files in my PATH, but my path looks like it got into some infinite loop.
This is how it looks in the integrated terminal:
<img width="930" alt="image" src="https://user-images.githubusercontent.com/146173/110469505-675c5280-80e2-11eb-9d38-1252778f1b23.png">
As opposed to iTerm:
<img width="977" alt="image" src="https://user-images.githubusercontent.com/146173/110469593-82c75d80-80e2-11eb-8b1e-3fdf5121d37e.png">

My shell is fish, but I got exactly the same behavior with zsh.

**Comment by omeraloni** [comment]
@roblourens this is the error I get when trying to attach to a container:
<img width="281" alt="image" src="https://user-images.githubusercontent.com/146173/110471079-5b719000-80e4-11eb-87b0-65b0634902b6.png">
<img width="848" alt="image" src="https://user-images.githubusercontent.com/146173/110471121-69bfac00-80e4-11eb-966d-944ca09db8b0.png">

Obviously Docker is running (connected to contained in 1.53.2).

**Comment by roblourens** [comment]
We just released an update to VS Code Insiders that has this fix. Please try it out and let me know how it goes.

**Comment by leandroyalet** [comment]
> Can you clarify what you changed in your PATH?

@roblourens sorry for the delay, I removed old and missing paths

**Comment by omeraloni** [comment]
> We just released an update to VS Code Insiders that has this fix. Please try it out and let me know how it goes.

Works for me, thanks for the quick fix!

**Comment by roblourens** [comment]
🎉 and we just released this to stable as well in 1.54.2
